### PR TITLE
Fix shellcheck failure in cluster/gce/util.sh (eliminate SC2155)

### PR DESCRIPTION
**Pull request by k-toyoda-pi**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fix a part of shellcheck failure in cluster/gce/util.sh.

`shellcheck` detects many many failures in `cluster/gce/util.sh`.
This PR fixes only SC2155 in that file. Because I think it makes us easier to understand the modifications.

See the below for the detail of SC2155.
```
SC2155: Declare and assign separately to avoid masking return values.
```
- https://github.com/koalaman/shellcheck/wiki/SC2155

**Which issue(s) this PR fixes**:
Ref #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -2332,7 +2342,8 @@ function check-existing() {
 }
 
 function check-network-mode() {
-  local mode="$(gcloud compute networks list --filter="name=('${NETWORK}')" --project ${NETWORK_PROJECT} --format='value(x_gcloud_subnet_mode)' || true)"
+  local mode
+  mode="$(gcloud compute networks list --filter="name=('${NETWORK}')" --project ${NETWORK_PROJECT} --format='value(x_gcloud_subnet_mode)' || true)"
   # The deprecated field uses lower case. Convert to upper case for consistency.
   echo "$(echo $mode | tr [a-z] [A-Z])"
```
Comment (BenTheElder) [inline cluster/gce/util.sh]: this is a pretty useless use of echo isn't it?

**Comment by BenTheElder** [comment]
i'm not sure how many of these we want to move to splitting off local to another line to just get the same behavior by masking the return code with `|| true`. otoh, i'm a bit scared to change the behavior of this script and break lots of things..

**Comment by k-toyoda-pi** [comment]
Thank you for the comment.
I think that the original code expects it returns `true`. Because `local foo=$(bar)` always returns `true`.
I checked the behavior around the code and added `|| true` on some of the places.
But, to make it safer, we should add `|| true` or `shellcheck disable` to the all detected lines.

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -741,7 +742,8 @@ function construct-common-kubelet-flags {
 # $1: if 'true', we're rendering flags for a master, else a node
 function construct-linux-kubelet-flags {
   local master="$1"
-  local flags="$(construct-common-kubelet-flags)"
+  local flags
+  flags="$(construct-common-kubelet-flags)"
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: Quotes are superfluous in assignments like this. Try it out:

$ a=$(echo foo bar)           
$ echo $a
foo bar
$

File: cluster/gce/util.sh [hunk 3]
```diff
@@ -787,7 +789,8 @@ function construct-linux-kubelet-flags {
     flags+=" --non-masquerade-cidr=${NON_MASQUERADE_CIDR}"
   fi
   flags+=" --volume-plugin-dir=${VOLUME_PLUGIN_DIR}"
-  local node_labels="$(build-linux-node-labels ${master})"
+  local node_labels
+  node_labels="$(build-linux-node-labels ${master})"
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: See previous comment on quotes in assignments like this.

File: cluster/gce/util.sh [hunk 4]
```diff
@@ -810,12 +813,14 @@ function construct-linux-kubelet-flags {
 # binPath parameter, and single-quotes get parsed as characters instead of
 # string delimiters.
 function construct-windows-kubelet-flags {
-  local flags="$(construct-common-kubelet-flags)"
+  local flags
+  flags="$(construct-common-kubelet-flags)"
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: Same

File: cluster/gce/util.sh [hunk 5]
```diff
@@ -810,12 +813,14 @@ function construct-linux-kubelet-flags {
...
 
   # Note: NODE_KUBELET_TEST_ARGS is empty in typical kube-up runs.
   flags+=" ${NODE_KUBELET_TEST_ARGS:-}"
 
-  local node_labels="$(build-windows-node-labels)"
+  local node_labels
+  node_labels="$(build-windows-node-labels)"
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: Same

File: cluster/gce/util.sh [hunk 6]
```diff
@@ -1478,7 +1483,8 @@ EOF
           # TODO(kubernetes/autoscaler#718): AUTOSCALER_ENV_VARS is a hotfix for cluster autoscaler,
           # which reads the kube-env to determine the shape of a node and was broken by #60020.
           # This should be removed as soon as a more reliable source of information is available!
-          local node_labels="$(build-linux-node-labels false)"
+          local node_labels
+          node_labels="$(build-linux-node-labels false)"
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: Same

File: cluster/gce/util.sh [hunk 7]
```diff
@@ -1878,7 +1884,8 @@ function generate-konnectivity-server-certs {
 # $1 master env (kube-env of master; result of calling get-master-env)
 # $2 env key to use
 function get-env-val() {
-  local match=`(echo "${1}" | grep -E "^${2}:") || echo ""`
+  local match
+  match=`(echo "${1}" | grep -E "^${2}:") || echo ""`
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: Please do not use legacy backticks, but use contemporary $() instead.

https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xcu_chap02.html#tag_23_02_06_03

File: cluster/gce/util.sh [hunk 8]
```diff
@@ -2332,7 +2342,8 @@ function check-existing() {
 }
 
 function check-network-mode() {
-  local mode="$(gcloud compute networks list --filter="name=('${NETWORK}')" --project ${NETWORK_PROJECT} --format='value(x_gcloud_subnet_mode)' || true)"
+  local mode
+  mode="$(gcloud compute networks list --filter="name=('${NETWORK}')" --project ${NETWORK_PROJECT} --format='value(x_gcloud_subnet_mode)' || true)"
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: Superfluous quotes for this type of assignment.

$ a=$(echo foo bar)           
$ echo $a
foo bar
$

File: cluster/gce/util.sh [hunk 9]
```diff
@@ -2809,7 +2822,8 @@ function replicate-master() {
     --type "${MASTER_DISK_TYPE}" \
     --size "${MASTER_DISK_SIZE}"
 
-  local existing_master_replicas="$(get-all-replica-names)"
+  local existing_master_replicas
+  existing_master_replicas="$(get-all-replica-names)"
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: Superfluous quotes.

File: cluster/gce/util.sh [hunk 10]
```diff
@@ -2867,8 +2882,10 @@ function create-loadbalancer() {
     return
   fi
 
-  local EXISTING_MASTER_NAME="$(get-all-replica-names)"
-  local EXISTING_MASTER_ZONE=$(gcloud compute instances list "${EXISTING_MASTER_NAME}" \
+  local EXISTING_MASTER_NAME
+  EXISTING_MASTER_NAME="$(get-all-replica-names)"
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: Same

File: cluster/gce/util.sh [hunk 11]
```diff
@@ -2951,8 +2970,10 @@ function create-internal-loadbalancer() {
...
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: Same

File: cluster/gce/util.sh [hunk 12]
```diff
@@ -3542,17 +3568,20 @@ function kube-down() {
...
+    --format "value(zone)" | wc -l || true)
 
   # In the replicated scenario, if there's only a single master left, we should also delete load balancer in front of it.
   if [[ "${REMAINING_MASTER_COUNT}" -eq 1 ]]; then
     if gcloud compute forwarding-rules describe "${MASTER_NAME}" --region "${REGION}" --project "${PROJECT}" &>/dev/null; then
       detect-master
-      local REMAINING_REPLICA_NAME="$(get-all-replica-names)"
-      local REMAINING_REPLICA_ZONE=$(gcloud compute instances list "${REMAINING_REPLICA_NAME}" \
+      local REMAINING_REPLICA_NAME
+      REMAINING_REPLICA_NAME="$(get-all-replica-names)"
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: Same

File: cluster/gce/util.sh [hunk 13]
```diff
@@ -3888,7 +3919,8 @@ function test-setup() {
 
   # Open up port 80 & 8080 so common containers on minions can be reached
   # TODO(roberthbailey): Remove this once we are no longer relying on hostPorts.
-  local start=`date +%s`
+  local start
+  start=`date +%s`
```
Comment (joakimr-axis) [inline cluster/gce/util.sh]: start=$(date +%s)

**Pull request closed without merging**